### PR TITLE
[ENH] Add a pre-commit check to prevent urls pointing to pymc docs

### PR DESCRIPTION
**Pull request by juanitorduz**
Fixes https://github.com/pymc-devs/pymc-examples/issues/292

I would support by solving this issue. However from the description I am not 100% the cope of which subfolders I need to add. I am taking the implementation from https://github.com/pymc-devs/pymc/pull/5486 as suggested in the issue description. 

Once I get some initial feedback I will keep working on this one :)

### CODE REVIEW

**Comment by OriolAbril** [comment]
Thanks! It should check all the files inside the `examples/` folder except the `conf.py` file.

**Comment by juanitorduz** [comment]
@OriolAbril is https://github.com/pymc-devs/pymc-examples/pull/311/commits/dd3c1380b071bafca56e8c22e3d891b59cbf850f enough or should add / remove websites from the `entry: >` specification?

File: .pre-commit-config.yaml [hunk 1]
```diff
@@ -55,6 +55,27 @@ repos:
       minimum_pre_commit_version: 2.8.0
       name: Check notebooks have watermark (see Jupyter style guide from PyMC docs)
       types: [jupyter]
+    - id: no-internal-links
+      name: Check no internal links are in the docs
+      description: >-
+        'A quick check to prevent urls pointing to pymc docs'
+      files: ^examples/
+      exclude: >
```
Comment (OriolAbril) [inline .pre-commit-config.yaml:63]: The 404.md file should also be excluded. And for now I would exclude all the notebooks that fail the check too so we can merge the PR and then work on fixing the issues

File: .pre-commit-config.yaml [hunk 2]
```diff
@@ -55,6 +55,27 @@ repos:
...
+          (?x)(conf.py)
+      entry: >
+          (?x)(arviz-devs.github.io|
+               aesara.readthedocs.io|
+               aeppl.readthedocs.io|
+               pymc-experimental.readthedocs.io|
+               docs.pymc.io|
+               numpy.org/doc|
+               pymc-examples.readthedocs.io|
+               myst-parser.readthedocs.io|
```
Comment (OriolAbril) [inline .pre-commit-config.yaml:73]: The two myst ones can be removed, but `python.arviz.org` and `docs.xarray.dev` should be added

Comment (juanitorduz) [reply .pre-commit-config.yaml]: Added in https://github.com/pymc-devs/pymc-examples/pull/311/commits/22e83a6e6c6f1b3f7e8ac70c767422268420b6af

File: .pre-commit-config.yaml (hunk 1 above)
Comment (juanitorduz) [reply .pre-commit-config.yaml:63]: `404.md` added in https://github.com/pymc-devs/pymc-examples/pull/311/commits/22e83a6e6c6f1b3f7e8ac70c767422268420b6af

Comment (juanitorduz) [reply .pre-commit-config.yaml:63]: I added all the notebooks which were failing and I also had to add `index.md` to that the tests pass.

**Comment by OriolAbril** [comment]
Thanks! Can you check if any of the now temporarily excluded notebooks is also in the "Done" column of the tracker project? https://github.com/pymc-devs/pymc-examples/projects/1

**Comment by juanitorduz** [comment]
Out of the 13 tickets in the "Done" column I do not see any which is also in this exclusion list.

**Review by OriolAbril — APPROVED** [review]
> Out of the 13 tickets in the "Done" column I do not see any which is also in this exclusion list.

Great, thanks for the PR, I think it is ready to merge

**Pull request merged**